plugin.rb

```diff
@@ -996,6 +1019,39 @@ def show
       render_serialized query, DataExplorer::QuerySerializer, root: 'query'
     end
 
+    def group_reports_index
+      respond_to do |format|
+        format.html { render 'groups/show' }
+        format.json do        
+          group = Group.find_by(name: params["group_name"])
+          return render(status: 404, json: {}) unless current_user && user_included_in_group(group)
+
+          queries = DataExplorer::Query.all
+          queries.select! { |query| query.group_ids.include?(group.id.to_s) }
+          render_serialized queries, DataExplorer::QuerySerializer, root: 'queries'
+        end
+      end
+    end
+
+    def group_reports_show
+      respond_to do |format|
+        format.html { render 'groups/show' }
+        format.json do    
+          query = DataExplorer::Query.find(params[:id].to_i)
+          return render(status: 404, json: {}) unless current_user && user_can_access_query(query)
```

Comment:
a guardian check at the entry is what we want here, no to scope it down to json only. the html should 404 as well.